### DISCUSSION TITLE
How to get lang in functions

### DISCUSSION THREAD

**Discussion by lynn2910** [MathieuTricoire/l10n#3] [Q&A] [answered]
I'm trying to build a function "DATETIME" that can format the date, but i didn't find how to get the language used (because, for exemple, in france we saw `22/02/2023` but in another countries they saw  `02/22/2023`)

How to get the language in a function ? (that we register at init)

**Comment by MathieuTricoire** [comment, marked as answer]
You need to create a new type that implements the `FluentType` trait and add a function that will return a `FluentValue::Custom` with this new type, then you'll be able to format your value with the locale, everything here come from `fluent-bundle`.

There is an example here: https://github.com/MathieuTricoire/l10n-examples/blob/79fcdc75e3e7e52c7d5689710aad839d9ef1b3ac/src/fluent/time.rs (I have updated it because it was not really relevant until now, you can clone the repo and run it)

- `FluentValue` enum: https://docs.rs/fluent-bundle/0.15.2/fluent_bundle/enum.FluentValue.html
- `FluentType` trait: https://docs.rs/fluent-bundle/0.15.2/fluent_bundle/types/trait.FluentType.html
